### PR TITLE
[ADF-4579] Layout - RTL support

### PR DESCRIPTION
**Pull request by pionnegru**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
[ADF-4579](https://issues.alfresco.com/jira/browse/ADF-4579)

**What is the new behaviour?**

- dynamically add document direction 
- layout direction not depended on demoshell logic
- material components apply direction correctly 

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: demo-shell/src/app/components/app-layout/app-layout.component.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
-<adf-sidenav-layout [sidenavMin]="70" [sidenavMax]="220" [stepOver]="780" [hideSidenav]="hideSidenav" [direction]="direction"
-    [expandedSidenav]="expandedSidenav" (expanded)="setState($event)" [position]="position">
+<adf-sidenav-layout [sidenavMin]="70" [sidenavMax]="220" [stepOver]="780" [hideSidenav]="hideSidenav"
+    [expandedSidenav]="expandedSidenav" (expanded)="setState($event)" [position]="position" [dir]="direction">
```
Comment (DenysVuika) [inline demo-shell/src/app/components/app-layout/app-layout.component.html]: component should detect/use direction itself, without demo shell

**Comment by davidcanonieto** [comment]
Tests failing 
FAILED TESTS:
  AuthenticationService
    when the setting is BPM
      ✖ [BPM] should be able to retrieve current logged in user
        Chromium 62.0.3202 (Ubuntu 0.0.0)
      Uncaught QuotaExceededError: Failed to execute 'setItem' on 'Storage': Setting the value of 'GUEST__supportedPageSizes' exceeded the quota. thrown
  StorageService
    ✖ should set a property with the prefix in the local storage
      Chromium 62.0.3202 (Ubuntu 0.0.0)
    Expected 'ADF_APP_GUEST__paginationSize' to be 'ADF_APP_test_key'.
        at services/storage.service.spec.ts:70:32
        at ZoneDelegate.push.../../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (home/travis/build/Alfresco/[secure]-ng2-components/node_modules/zone.js/dist/zone.js:391:1)
        at ProxyZoneSpec.push.../../node_modules/zone.js/dist/zone-testing.js.ProxyZoneSpec.onInvoke (home/travis/build/Alfresco/[secure]-ng2-components/node_modules/zone.js/dist/zone-testing.js:289:1)
        at ZoneDelegate.push.../../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (home/travis/build/Alfresco/[secure]-ng2-components/node_modules/zone.js/dist/zone.js:390:1)
    ✖ should set a property without a prefix in the local storage
      Chromium 62.0.3202 (Ubuntu 0.0.0)
    Expected 'GUEST__paginationSize' to be 'test_key'.
        at services/storage.service.spec.ts:84:32
        at ZoneDelegate.push.../../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (home/travis/build/Alfresco/[secure]-ng2-components/node_modules/zone.js/dist/zone.js:391:1)
        at ProxyZoneSpec.push.../../node_modules/zone.js/dist/zone-testing.js.ProxyZoneSpec.onInvoke (home/travis/build/Alfresco/[secure]-ng2-components/node_modules/zone.js/dist/zone-testing.js:289:1)
        at ZoneDelegate.push.../../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (home/travis/build/Alfresco/[secure]-ng2-components/node_modules/zone.js/dist/zone.js:390:1)
  UserPreferencesService
    ✖ should get default pagination from app config
      Chromium 62.0.3202 (Ubuntu 0.0.0)
    Expected 25 to be 0.
        at services/user-preferences.service.spec.ts:65:48
        at ZoneDelegate.push.../../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (home/travis/build/Alfresco/[secure]-ng2-components/node_modules/zone.js/dist/zone.js:391:1)
        at ProxyZoneSpec.push.../../node_modules/zone.js/dist/zone-testing.js.ProxyZoneSpec.onInvoke (home/travis/build/Alfresco/[secure]-ng2-components/node_modules/zone.js/dist/zone-testing.js:289:1)
        at ZoneDelegate.push.../../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (home/travis/build/Alfresco/[secure]-ng2-components/node_modules/zone.js/dist/zone.js:390:1)

**Comment by pionnegru** [comment]
not touched this service in this PR. I have to investigate

**Comment by eromano** [comment]
Something seems to be failing when I change in the config editor from ltr to rtl

**Comment by pionnegru** [comment]
@eromano  can you be more specific?

**Pull request merged**